static/app/views/starfish/views/spans/spansView.tsx

```diff
@@ -75,6 +76,28 @@ export default function SpansView(props: Props) {
     enabled: groupIDs.length > 0,
   });
 
+  // Initialize the selected span group if it exists in the URL
+  const {onSelect} = props;
+  const {query} = useLocation();
+  const selectedSpanGroup = query.group_id;
+  const [initializedSelectedSpan, setInitializedSelectedSpan] = useState(false);
+  useEffect(() => {
+    if (
+      !initializedSelectedSpan &&
+      !areSpansLoading &&
+      selectedSpanGroup &&
+      spansData.length > 0
+    ) {
+      const selectedSpanData = spansData.find(
+        ({group_id}) => group_id === selectedSpanGroup
+      );
+      if (selectedSpanData) {
+        onSelect(mapRowKeys(selectedSpanData, selectedSpanData.span_operation));
+      }
+      setInitializedSelectedSpan(true);
+    }
+  }, [areSpansLoading, initializedSelectedSpan, onSelect, selectedSpanGroup, spansData]);
+
```

Comment:
I'd prefer to move this logic to the parent `index.tsx`, but this is the easiest way to implement and test this logic right now.

This code can also probably be turned into a reusable hook for the other modules in a future update

Reply:
Whoa, this is really complicated! It does the job fine, but it might be simpler at this point to update `SpanSummaryPanel` to accept a group ID instead of a span object. Then panel can load the span op, description, and action itself, and can accept the group ID straight from the URL.

Or, if you really don't want to load the span information in the panel that way, maybe the parent component can optionally load the information from URL _or_ accept the span information in the `onSelect` callback from this table, and then you wouldn't need the `useEffect` somersault

Reply:
Yeah, I think ideally the panel should be self sufficient and query on its own. Going to merge for now, but this will be my planned refactor for an upcoming pr when I get around to it!